lib/ex_uid2/encryption/identity.ex

```diff
@@ -38,4 +48,10 @@ defmodule ExUid2.Encryption.Identity do
   end
 
   def parse(_), do: {:error, :invalid_identity_payload}
+
+  @spec make_envelope(t()) :: binary()
+  def make_envelope(identity) do
+    <<identity.site_id::big-integer-32, identity.id_len::big-integer-32, identity.id_bin::binary,
+      0::big-integer-32, identity.established_ms::big-integer-64>>
```

Comment:
As nice and compact as this is, there is a slight performance penalty to repeatedly calling `struct.field`. Usually once I do more than 2 `struct.field` accesses in a single function, I switch to pattern matching instead.

Reply:
That's a good point. This function is only used in tests, but that's no excuse!. Moved the lookup to the function head.